Xamarin.Forms.Core/Picker.cs

```diff
@@ -150,8 +150,10 @@ void OnItemDisplayBindingChanged(BindingBase oldValue, BindingBase newValue)
 
 		void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 		{
-			SelectedIndex = SelectedIndex.Clamp(-1, Items.Count - 1);
-			UpdateSelectedItem();
```

Comment:
if the old and the new selectedIndex are different, setting the `SelectedIndex` will trigger the UpdateSelectedItem at the right time.